### PR TITLE
Rework Checkpoint interface and CheckpointHook/CheckpointHookFactory …

### PR DESCRIPTION
**Pull request by anjumfatima90**

### CODE REVIEW

**Comment by anjumfatima90** [comment]
#build

File: dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHookFactory.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * A checkpoint hook factory provides an instance of a checkpoint hook
+ * that will be used for a checkpoint and restore operation.
+ *
+ * @see Checkpoint#snapshot(Phase, java.io.File)
+ * @see CheckpointHook
+ */
+@FunctionalInterface
+public interface CheckpointHookFactory {
+    /**
+     * Phase which a snapshot is being taken.
```
Comment (tjwatson) [inline dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHookFactory.java]: Phase which a checkpoint is being taken.

File: dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHookFactory.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+     *
+     */
+    public enum Phase {
+        FEATURES,
```
Comment (tjwatson) [inline dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHookFactory.java:33]: Add the following javadoc to the FEATURES enum:
```java
       /**
         * Phase of startup after all feature bundles have been started and before
         * starting any configured applications
         */

```

File: dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHookFactory.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        APPLICATIONS;
```
Comment (tjwatson) [inline dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHookFactory.java:38]: Add the following javadoc to the APPLICATIONS enum:
```java
        /**
         * Phase of startup after all configured applications have been started
         * or have timed out starting.  No ports are opened yet
         */

```

File: dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHookFactory.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+         */
+        public static Phase getPhase(String p) {
+            return phases.get(p.trim().toUpperCase());
+        }
+    }
+
+    /**
+     * Creates a checkpoint hook for the specified phase.
+     * The instance of this hook will be used to prepare
+     * and restore a snapshot.
```
Comment (tjwatson) [inline dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHookFactory.java]: ... and restore a checkpoint.

File: dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHookFactory.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+     *
+     * @param phase the phase of the snapshot
```
Comment (tjwatson) [inline dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHookFactory.java]: ... the phase of the checkpoint

File: dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHook.java [hunk 6]
```diff
@@ -12,7 +12,7 @@
 
 import org.osgi.annotation.versioning.ConsumerType;
 
-import io.openliberty.checkpoint.spi.Checkpoint.Phase;
+import io.openliberty.checkpoint.spi.CheckpointHookFactory.Phase;
 
 /**
  * A hook that gets called during a snapshot to allow the system to
```
Comment (tjwatson) [inline dev/com.ibm.ws.kernel.boot.core/src/io/openliberty/checkpoint/spi/CheckpointHook.java]: replace all javadoc occurrences of `snapshot` with `checkpoint`

**Comment by anjumfatima90** [comment]
#build

**Review by tjwatson — APPROVED** [review]

**Pull request merged**